lib/ansible/modules/cloud/amazon/aws_codedeploy_application.py

```diff
@@ -0,0 +1,438 @@
+    required: false
+  alarm_configuration:
+    description:
+      - Information to add about Amazon CloudWatch alarms when the deployment group is created.
+    required: false
+  auto_rollback_config:
+    description:
+      - Configuration information for an automatic rollback that is added when a deployment group is created.
+    required: false
+  bg_deployment_config:
+    description:
+      - Information about blue/green deployment options for a deployment group.
+    required: false
+  load_balance_info:
+    description:
+      - Information about the load balancer used in a deployment.
+    required: false
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Create a CodeDeploy application with a deployment group and an IN_PLACE deployment style.
+- aws_codedeploy_application:
+  application_name: TestApplication
+  deployment_group: test_application_group
+  deployment_style:
+    deploymentType: "IN_PLACE"
+    deploymentOption: "WITHOUT_TRAFFIC_CONTROL"
+  service_role_arn: "arn:aws:iam::aws:policy/service-role/AWSCodeDeployRole"
+
+- aws_codedeploy_application:
+  application_name: TestApplication
+  deployment_group: test_application_group
+  deployment_style:
+    deploymentType: "IN_PLACE"
+    deploymentOption: "WITHOUT_TRAFFIC_CONTROL"
+  service_role_arn: "arn:aws:iam::aws:policy/service-role/AWSCodeDeployRole"
+  auto_scaling_groups: Test-Application-ASG
+  deployment_config: CodeDeployDefault.AllAtOnce
+
+- aws_codedeploy_application:
+  application_name: TestApplication
+  new_application_name: TestApplicationV1
+  deployment_group: test_application_group
+  new_deployment_group: test_application_v1
+  deployment_style:
+    deploymentType: "IN_PLACE"
+    deploymentOption: "WITH_TRAFFIC_CONTROL"
+  service_role_arn: "arn:aws:iam::aws:policy/service-role/AWSCodeDeployRole"
+  deployment_config: CodeDeployDefault.OneAtATime
+  load_balance_info:
+    elbInfoList:
+      - MyClassicLoadBalancer
+'''
+
+RETURN = '''
+application:
+  description: Information regarding the CodeDeploy application.
+  type: complex
+  returned: success
+  contains:
+    application_id:
+      description: The application ID.
+      type: string
+      sample: "81293494"
+    application_name:
+      description: The application name.
+      type: string
+      sample: "TestApplication"
+    create_time:
+      description: The time at which the application was created.
+      type: datetime
+      sample: 2017-05-30 010:00:00-05:00
+    linked_to_github:
+      description: True if the user has authenticated with GitHub for the specified application; otherwise, false.
+      type: boolean
+      sample: false
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, get_aws_connection_info, ec2_argument_spec, camel_dict_to_snake_dict
```

Comment:
Once you use `module.client`, `boto3_conn` and `get_aws_connection_info` are no longer required